### PR TITLE
Feature/aggregation loader

### PR DESCRIPTION
**Pull request by GustavoLastra**
1. Installs Spin.js v4 node module in frontend.
2. implements a shared component for the spinner.
3. implements a  shared service for the spinner
- defines a shared subject to control the spinner from another angular service
4. adds spinner to aggregation module

### CODE REVIEW

File: frontend/src/app/modules/aggregation/aggregation.module.ts [hunk 1]
```diff
@@ -8,13 +8,14 @@ import {AggregationChartComponent} from './components/aggregation-chart/aggregat
...
 import {FormsModule} from "@angular/forms";
+import {SpinnerComponent} from '../shared/components/spinner/spinner.component';
 
 const AggregationRoutes: Routes = [
   {path: 'show', component: AggregationComponent},
 ];
 
 @NgModule({
-  declarations: [AggregationComponent, AggregationChartComponent],
+  declarations: [AggregationComponent, AggregationChartComponent, SpinnerComponent],
```
[the remark this answers is not in the document]
Comment (BennyLi) [reply frontend/src/app/modules/aggregation/aggregation.module.ts]: This even results in an error in the console in my browser:

```
Error: Uncaught (in promise): Error: Type SpinnerComponent is part of the declarations of 2 modules: SharedModule and AggregationModule!
```

File: frontend/src/styles/cards.scss [hunk 2]
```diff
@@ -11,3 +11,8 @@
     margin: 0 0 20px 0;
   }
 }
+
```
[the remark this answers is not in the document]
Comment (GustavoLastra) [reply frontend/src/styles/cards.scss]: checked

File: frontend/src/app/modules/shared/components/spinner/spinner.component.scss [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+@keyframes spinner-line-fade-default {
+  0%, 100% {
+    opacity: 0.22; /* minimum opacity */
+  }
+  1% {
+    opacity: 1;
+  }
+}
```
[the remark this answers is not in the document]
Comment (GustavoLastra) [reply frontend/src/app/modules/shared/components/spinner/spinner.component.scss]: awesome!

File: frontend/src/app/modules/shared/components/spinner/spinner.component.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  ngOnDestroy() {
+    this.subscription.unsubscribe();
+  }
+
+  startSpinner() {
+    this.show = true;
+    this.spinner.spin(this.element.firstChild);
+    this.element.parentElement.classList.add( "parentSpinner");
```
[the remark this answers is not in the document]
Comment (GustavoLastra) [reply frontend/src/app/modules/shared/components/spinner/spinner.component.ts]: I agree

File: frontend/src/app/modules/aggregation/models/aggregation-chart-data.model.ts [hunk 5]
```diff
@@ -1,4 +1,6 @@
 import {AggregationChartSeries} from "./aggregation-chart-series.model";
+import {BehaviorSubject} from "rxjs";
+import {Loading} from "../../../models/loading.model";
```
[the remark this answers is not in the document]
Comment (GustavoLastra) [reply frontend/src/app/modules/aggregation/models/aggregation-chart-data.model.ts]: Not anymore

File: frontend/src/app/modules/aggregation/services/aggregation-chart-data.service.ts [hunk 6]
```diff
@@ -93,20 +94,24 @@ export class AggregationChartDataService {
 
   barchartAverageData$: BehaviorSubject<any> = new BehaviorSubject<any>([]);
   barchartMedianData$: BehaviorSubject<any> = new BehaviorSubject<any>([]);
+  isLoading$: BehaviorSubject<any> = new BehaviorSubject<any>(true);
```
[the remark this answers is not in the document]
Comment (GustavoLastra) [reply frontend/src/app/modules/aggregation/services/aggregation-chart-data.service.ts]: This variable is not needed anymore

File: frontend/src/app/modules/aggregation/services/aggregation-chart-data.service.ts [hunk 7]
```diff
@@ -171,7 +178,8 @@ export class AggregationChartDataService {
           highlighted: false,
           selected: false,
           hasComparative: false,
-          color: ''
+          color: '',
+          /*isLoading: false*/
```
[the remark this answers is not in the document]
Comment (GustavoLastra) [reply frontend/src/app/modules/aggregation/services/aggregation-chart-data.service.ts]: Yes

File: frontend/src/app/modules/aggregation/services/aggregation-chart-data.service.ts [hunk 8]
```diff
@@ -116,6 +121,8 @@ export class AggregationChartDataService {
       URL.AGGREGATION_BARCHART_DATA
     ).subscribe(result => {
       this.barchartMedianData$.next(this.sortDataByMeasurandOrder(result));
+      /*this.isLoading$.next(false);*/
```
[the remark this answers is not in the document]
Comment (GustavoLastra) [reply frontend/src/app/modules/aggregation/services/aggregation-chart-data.service.ts]: Yes

File: frontend/src/app/modules/shared/components/spinner/spinner.component.html [hunk 9]
```diff
@@ -0,0 +1,2 @@
+<div [style.display]="show == true ? 'block' : 'none'"></div>
+<!--<div *ngIf="show"></div>-->
```
[the remark this answers is not in the document]
Comment (GustavoLastra) [reply frontend/src/app/modules/shared/components/spinner/spinner.component.html]: Yes

File: frontend/src/app/modules/shared/services/spinner.service.spec.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import { TestBed } from '@angular/core/testing';
+
+import { SpinnerService } from './spinner.service';
+
+describe('SpinnerService', () => {
```
[the remark this answers is not in the document]
Comment (GustavoLastra) [reply frontend/src/app/modules/shared/services/spinner.service.spec.ts]: done

[the remark this answers is not in the document]
Comment (BennyLi) [reply frontend/src/app/modules/shared/services/spinner.service.spec.ts]: I ment to delete the whole file :)
I'll do it :+1:

**Pull request merged**